hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/DefaultStringifier.java

```diff
@@ -158,6 +158,9 @@ public static <K> K load(Configuration conf, String keyName,
   public static <K> void storeArray(Configuration conf, K[] items,
       String keyName) throws IOException {
 
+    if (items.length == 0) {
+      throw new IndexOutOfBoundsException();
+    }
     DefaultStringifier<K> stringifier = new DefaultStringifier<K>(conf, 
         GenericsUtil.getClass(items[0]));
```

Comment:
won't this trigger the same exception automatically? what does the test do before the new lines are added

Reply:
Thanks @steveloughran for the review.
Yes, the original code would trigger ArrayIndexOutOfBoundsException automatically, and the new test passes without the new lines.
But we found the same exception has been handled in other places in the same module and hence, request to handle it.

Reply:
ok